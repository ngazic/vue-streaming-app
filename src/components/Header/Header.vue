<template>
<header id="header">
  <div class="logo">
    <router-link to="/">Vue - Streaming App</router-link>
  </div>
  <nav>
    <ul>
      <li v-if="!auth">
        <router-link to="/signup" class="login">Sign Up</router-link>
      </li>
      <li v-if="!auth">
        <router-link to="/signin" class="login">Sign In</router-link>
      </li>
      <li v-if="auth">
        <button @click="onLogout" class="logout">Logout</button>
      </li>
    </ul>
  </nav>
</header>
</template>

<script lang="ts">
import Vue from 'vue'
export default Vue.extend({
  computed: {
    auth(): void {
      return this.$store.getters.isAuthenticated;
    }
  },
  methods: {
    onLogout() {
      this.$store.dispatch("logout");
    }
  }
});
</script>

<style scoped>
#header {
  height: 56px;
  display: flex;
  flex-flow: row;
  justify-content: space-between;
  align-items: center;
  background-color: #521751;
  padding: 0 20px;
}

.logo {
  font-weight: bold;
  color: white;
}

.logo a {
  text-decoration: none;
  color: white;
}

@media only screen and (max-width: 400px) {
  .logo a {
    display: block;
    text-overflow: ellipsis;
    white-space: nowrap;
    max-width: 80px;
    overflow: hidden;
  }
}

nav {
  height: 100%;
}

ul {
  list-style: none;
  margin: 0;
  padding: 0;
  height: 100%;
  display: flex;
  flex-flow: row;
  align-items: center;
}

li {
  margin: 0 16px;
}

li a {
  text-decoration: none;
  color: white;
}

li a:hover,
li a:active,
li a.router-link-active {
  color: #fa923f;
}

.logout {
  background-color: transparent;
  border: none;
  font: inherit;
  color: white;
  cursor: pointer;
}
</style>
