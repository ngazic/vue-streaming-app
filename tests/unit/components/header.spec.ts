import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import Vuex from 'vuex'
import Header from '@/components/Header/Header.vue'

describe('Header.vue', () => {
  const localVue = createLocalVue()
  localVue.use(VueRouter)
  localVue.use(Vuex)

  describe('show/hide links on authentification', () => {
    const store = new Vuex.Store({
      getters: {
        isAuthenticated(): boolean {
          return true;
        }
      }
    })
    const wrapper = shallowMount(Header,
      {
        store,
        localVue,
      },
    )
    // for debbuging wrapper uncomment this line:
    // wrapper.overview()

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('if user is logged in don\'t show singin and signup buttons ', () => {
      expect(wrapper.find('.login').exists()).toBe(false)
    });

    it('should show logout button if a user is logged in ', () => {
      expect(wrapper.find('.logout').exists()).toBe(true)
    });
  });
})