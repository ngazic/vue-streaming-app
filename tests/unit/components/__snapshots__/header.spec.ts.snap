// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue show/hide links on authentification renders correctly 1`] = `
<header id="header">
  <div class="logo">
    <router-link-stub to="/" tag="a" ariacurrentvalue="page" event="click">Vue - Streaming App</router-link-stub>
  </div>
  <nav>
    <ul>
      <!---->
      <!---->
      <li><button class="logout">Logout</button></li>
    </ul>
  </nav>
</header>
`;
